src/py/aspen/config/config.py

```diff
@@ -58,26 +58,29 @@ class Config:
     _config_flask_properties: Optional[Set[str]] = None
     _subclasses: MutableMapping[str, Type[Config]] = dict()
 
-    def __init_subclass__(cls: Type[Config], descriptive_name: str, *args, **kwargs):
+    def __init_subclass__(
+        cls: Type[Config], descriptive_name: Optional[str] = None, *args, **kwargs
+    ):
         # the type: ignore on the next line is due to
         # https://github.com/python/mypy/issues/4660
         super().__init_subclass__(*args, **kwargs)  # type: ignore
-        Config._subclasses[descriptive_name] = cls
-
-        # get a list of @flaskproperties for Config and the subclass of Config, and
-        # compare to see if there are any attributes that were marked as
-        # @flaskproperty in Config but not in the subclass.
-        if Config._config_flask_properties is None:
-            Config._config_flask_properties = _get_flaskproperty_names(Config)
-        flask_properties = _get_flaskproperty_names(cls)
-
-        if not Config._config_flask_properties.issubset(flask_properties):
-            missing_properties = Config._config_flask_properties - flask_properties
-            missing_properties_string = ", ".join(missing_properties)
-            raise Exception(
-                f"Attributes ({missing_properties_string}) defined as flask properties"
-                f" in Config but not in {cls.__name__}"
-            )
+        if descriptive_name is not None:
+            Config._subclasses[descriptive_name] = cls
+
+            # get a list of @flaskproperties for Config and the subclass of Config, and
+            # compare to see if there are any attributes that were marked as
+            # @flaskproperty in Config but not in the subclass.
```

Comment:
don't ABC's do something similar to this?

Reply:
If you tag something as an abstract class (abc.abstractclass), or there are abstract methods (abc.abstractmethod), it cannot be instantiated.  This is similar in concept but we can't quite use those constructs to achieve the same.